name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/PostbankPDFExtractor.java

```diff
@@ -1,325 +1,330 @@
+                        .section("date", "time").optional()
+                        .match("^Schlusstag\\/-Zeit (?<date>\\d+.\\d+.\\d{4}) (?<time>\\d+:\\d+:\\d+).*")
+                        .assign((t, v) -> {
+                            if (v.get("time") != null)
+                                t.setDate(asDate(v.get("date"), v.get("time")));
+                            else
+                                t.setDate(asDate(v.get("date")));
+                        })
+
+                        // Ausmachender Betrag 9.978,18- EUR
+                        .section("amount", "currency")
+                        .match("^Ausmachender Betrag (?<amount>[.,\\d]+)[\\+-]? (?<currency>[\\w]{3})")
+                        .assign((t, v) -> {
+                            t.setAmount(asAmount(v.get("amount")));
+                            t.setCurrencyCode(v.get("currency"));
+                        })
+
+                        .wrap(BuySellEntryItem::new);
+
+        addTaxesSectionsTransaction(pdfTransaction, type);
+        addFeesSectionsTransaction(pdfTransaction, type);
+    }
+
+    private void addDividendeTransaction()
+    {
+        DocumentType type = new DocumentType("(Dividendengutschrift|Aussch.ttung Investmentfonds|Gutschrift von Investmentertr.gen|Ertragsgutschrift)");
+        this.addDocumentTyp(type);
+
+        Block block = new Block("(Dividendengutschrift|Aussch.ttung Investmentfonds|Gutschrift von Investmentertr.gen|Ertragsgutschrift.*)");
+        type.addBlock(block);
+        Transaction<AccountTransaction> pdfTransaction = new Transaction<AccountTransaction>().subject(() -> {
+            AccountTransaction entry = new AccountTransaction();
+            entry.setType(AccountTransaction.Type.DIVIDENDS);
+            return entry;
+        });
+
+        pdfTransaction
+
+                        // Stück 12 JOHNSON & JOHNSON SHARES US4781601046
+                        // (853260)
+                        // REGISTERED SHARES DL 1
+                        .section("isin", "wkn", "name", "shares", "name1")
+                        .match("^St.ck (?<shares>[\\d.,]+) (?<name>.*) (?<isin>[\\w]{12}.*) (\\((?<wkn>.*)\\).*)")
+                        .match("(?<name1>.*)") //
+                        .assign((t, v) -> {
+                            if (!v.get("name1").startsWith("Zahlbarkeitstag"))
+                                v.put("name", v.get("name").trim() + " " + v.get("name1"));
+
+                            t.setSecurity(getOrCreateSecurity(v));
+                            t.setShares(asShares(v.get("shares")));
+                        })
+                        
+                        // Zahlbarkeitstag 09.03.2021
+                        .section("date")
+                        .match("^Zahlbarkeitstag (?<date>\\d+.\\d+.\\d{4}).*")
+                        .assign((t,v) -> t.setDateTime(asDate(v.get("date"))))
+
+                        // Ex-Tag 22.02.2021 Art der Dividende Quartalsdividende
```

Comment:
could you not merge the `note` and the `exdate` here?

Reply:
There are 3 types of ex-date-lines in the dividend-text-files. I changed it to this and added the 'exdate' param to the note section.

`// Ex-Tag 22.02.2021
                        // Ex-Tag 15.04.2016 anteil pro St. 0,007344000 EUR
                        .section("exDate")
                        .match("^Ex-Tag (?<exDate>\\d+.\\d+.\\d{4}).*") 
                        .assign((t, v) -> {
                            t.setExDate(asDate(v.get("exDate")).toLocalDate());
                        })

                        // Ex-Tag 22.02.2021 Art der Dividende Quartalsdividende
                        .section("exDate", "note").optional()
                        .match("^Ex-Tag (?<exDate>\\d+.\\d+.\\d{4}) Art der Dividende (?<note>.*)")
                        .assign((t, v) ->{
                            t.setNote(v.get("note"));
                            t.setExDate(asDate(v.get("exDate")).toLocalDate());
                        }) `